Add display for user stats and styling for bg color & footer

pachliopta opened the pull request:
### Changes
Addresses #11 and adds some additional styling options.

* Add parameter `stats` to allow users to show their scrobbles.
* Add parameter `show_user` to allow users to show their profile information (username & picture).
* Add parameter `footer_stats` places the user stats and profile in the footer.
* Add parameter `footer_size` to allow users to show display a footer. `footer_stats` takes priority.
* Add parameter `bg_color` to allow users to customize the card's color with a RGB/A hex code.
* Adjusts styling of the card for clean rounded corners and slightly better alignment of title in compact mode.
* Updates docs for new API options

### Previews
Here are some previews. There's a lot of combinations introduced in this PR so I can't show them all.

|  description | preview (at width=300) |
| -------- | ------- |                     
| `api?stats=true&show_user=true`</br>user info | ![prev1](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/a97df69d-53a0-4f25-a200-269a64a0a96d) | 
| `api?bg_color=eebbed`</br>bg_color </br></br> supports transparency. a lot of colors are not very nice.| ![Screenshot 2023-11-29 073619](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/ce251611-93d9-4610-81be-b38ab0390925) |
| `api?show_user=true&footer_size=normal`</br>show_user (no stats) and default footer</br></br> the default footer image was created in after effects and is based off the official 'spiral wave' last.fm merch. its cropped to like 1% size and about 2x2px. | ![prev3](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/2ba45925-6506-4347-91b4-5da0bbb75568) |
|  `api?show_user=true&stats=true&footer_stats=true`</br>user info in footer | ![prev4](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/6c3e3de2-3807-4ca5-aa65-2b245af3b1d2) |
| `api?show_user=true&stats=true&header_size=compact`</br>compact styling | ![prev5](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/ba4fd7de-d060-4d84-ae1f-03e84b67f8e5) |

### Additional Notes
The API response for existing links will display the same as previously (excluding aforementioned style adjustments). The design of new elements are naturally a bit opinionated so it's completely open to change!

pachliopta committed:
feat: user stats, bg color, footer styling

JeffreyCA commented on README.md:
```diff
@@ -32,9 +32,14 @@ You can customize your list.fm status by adding query parameters after the url.
 | `count` | Number of recent tracks to display | number | 5 | 1 - 10 |
 | `width` | Width of the card in pixels | number | 400 | 300-1000 |
 | `loved` | Show a heart indicator for loved tracks | boolean | false | true, false |
+| `stats` | Show your scrobbles, arist count, and track count | boolean | false | true, false |
```
Sorry this will require some refactoring but I prefer combining the `stats` and `footer_stats` options into `header_size` and `footer_style`, something like:

| | |
|--|--|
| `header_size` (`header_style` is more fitting but keep for backcompat) | none, compact, normal, compact_stats, normal_stats (and a none_stats/stats_only if you want) |
| `footer_style` | none, wave, compact_stats, normal_stats |

pachliopta replied:
Totally agree, I made the URL quite unwieldy and I unfortunately that mirrored within the code. This made things much nicer though!

In addition to your proposed changes I added `normal_stats_only` and `compact_stats_only` to the header, and `compact_blank` and `normal_blank` to the footer. The blank footers allow the user information to be added to the footer independent of stats. I also chose to add header_style as an alias of sorts of header_size, maintaining backwards compatibility. Internally and in the docs, all except the alias is updated to style wording.

JeffreyCA commented on styles/svg.ts:
```diff
@@ -30,6 +34,52 @@ svg {
+    column-gap: 3px;
+    align-items: baseline;
+}
+.lastfm-stats > div > span:nth-child(2) {
```
The font size is a bit too small for my liking, let's bump it to around 11-12px? It'll affect the height calculations I think.

pachliopta replied:
Bumped it up to 11px. I think 12px threw off the balance slightly, but in retrospect I don't think I tried changing the size of the numbers too - should look fine if it needs to be any larger.

JeffreyCA commented on components/TrackList/Stats.tsx:
```diff
@@ -0,0 +1,42 @@
+                </div>
+                <div>
+                    <Text strong >{parseInt(props.userInfo.artist_count).toLocaleString()}</Text>
+                    <Text strong>Arists</Text>
```
```suggestion
                    <Text strong>Artists</Text>
```

pachliopta replied:
I must have scanned past it so many times, I can't believe I missed it. I'll blame my font size 😅

JeffreyCA commented:
This is awesome, thank you for the contribution! I left a few comments. Also can you run Prettier to clean up the formatting? The build won't succeed unless the code is formatted properly

pachliopta committed:
refactor: rework user stats styling

pachliopta commented:
Should have addressed these changes now! Did require a little bit of refactoring, but your proposed API also made for much more pleasant code.

Other than the requested changes, it largely looks the same as my initial table. Otherwise, the sizing calculations are tightened up and the user profile can be manipulated independently of the stats. The stats will also center themselves when they are the only item in their respective sections.

### API changes:
- removed `footer_stats`
- removed `stats`
- changed `footer_size` -> `footer_style`
- added `header_style` as an alias for `header_size`
    - backwards compatibility is maintained, though the docs are updated to reflect consistent use of `header_style`
- `footer_style` now accepts the following values:
    - `none` | `wave` | `normal_stats` | `compact_stats` | `normal` | `compact`
- `header_style` now accepts the following values:
    - `none` | `normal` | `compact` | `normal_stats` | `compact_stats` | `normal_stats_only` | `compact_stats_only`
- `show_user` now accepts the following values, giving control over its position:
    - `never` | `always` | `header` | `footer`

### Previews:
![image](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/b168b208-8560-439a-bbc2-39ce59086c0c)
![image](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/b6d59f34-173a-4976-8040-a2bf40e4b959)
![image](https://github.com/JeffreyCA/lastfm-recently-played-readme/assets/111627835/0ba975ec-f594-403d-ad77-fd69c8a5207a)

I applied prettier in a separate commit but messed up a rebase, sorry in advance for that! I touched most of what prettier did so the diffs are hopefully not too bad.

pachliopta committed:
docs: add user & stats styling usage

pachliopta committed:
fix: eslint issues

JeffreyCA commented:
Perfect, thanks again!
